test/units/config/test_manager.py

```diff
@@ -0,0 +1,44 @@
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import tempfile
+
+from ansible.compat.tests import unittest
+from ansible.module_utils.six.moves import configparser
+
+from ansible.config import manager
+
+
+class TestConfigManager(unittest.TestCase):
+    def tearDown(self):
+        os.unlink(self.tmp_file.name)
+
+    def _config_file(self, contents=None):
+
+        self.tmp_file = tempfile.NamedTemporaryFile(delete=False, suffix='.cfg')
+        ini_config = configparser.ConfigParser()
+        for item in contents:
+            ini_config.add_section(item['section'])
+            ini_config.set('defaults', item['name'], item['value'])
+
+        ini_config.write(self.tmp_file)
+        self.tmp_file.close()
+        return self.tmp_file
+
+    def test_hostfile(self):
```

Comment:
the tests should not really be for hostfile, as we are going to remove it.

a mock setting with 2 entries would test the functionality, not the specific setting